### PR TITLE
docs(CoC): update to 2.0

### PR DESCRIPTION
**Pull request by LuigiImVector**
## What does this PR do?
Improve repo

### Description

Apply #6792 and discuss the new version of CoC.

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: docs/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,50 +1,129 @@
...
+## Enforcement
 
 Instances of abusive, harassing, or otherwise unacceptable behavior may be
-reported by contacting a project maintainer at victorfelder at gmail.com. All
-complaints will be reviewed and investigated and will result in a response that
-is deemed necessary and appropriate to the circumstances. Maintainers are
-obligated to maintain confidentiality with regard to the reporter of an
-incident.
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Comment (LuigiImVector) [inline docs/CODE_OF_CONDUCT.md:64]: what to write here?

File: docs/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,50 +1,129 @@
...
+### 1. Correction
+
+**Community Impact**: Use of inappropriate language or other behavior deemed
+unprofessional or unwelcome in the community.
+
+**Consequence**: A private, written warning from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
+
+### 2. Warning
```
Comment (LuigiImVector) [inline docs/CODE_OF_CONDUCT.md:84]: Warning/Temporary Ban/Permanent ban are useless in github, do we remove them?

File: docs/CODE_OF_CONDUCT.md (hunk 1 above)
Comment (davorpa) [reply docs/CODE_OF_CONDUCT.md:64]: like in 1.3... the email contact

Comment (LuigiImVector) [reply docs/CODE_OF_CONDUCT.md:64]: I couldn't find the email

File: docs/CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,50 +1,129 @@
...
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
 
 Instances of abusive, harassing, or otherwise unacceptable behavior may be
-reported by contacting a project maintainer at victorfelder at gmail.com. All
```
Comment (davorpa) [inline docs/CODE_OF_CONDUCT.md:38]: @LuigiImVector, Here, at this line, is the latest email in format that bots can't spam too much

Comment (LuigiImVector) [reply docs/CODE_OF_CONDUCT.md:38]: oops, I totally didn't see it ahahah, thanks for the correction

**Comment by eshellman** [comment]
Thanks for presenting this in an actionable way. One thing that it surfaces is the community leadership issue - I _think_ the CoC was put in place before the transition to the Free Ebook Foundation - @vhf is still designated as the relevant contact. So we can't change this without either Victor's involvement or (much better) some more discussion about community leadership. Can we close this for now and start some discussion about how to implement it?

**Comment by LuigiImVector** [comment]
> Can we close this for now and start some discussion about how to implement it?

sure, I opened this pr to try to speed up this process

**Comment by LuigiImVector** [comment]
I forgot to close it

**Pull request closed without merging**